tests/pass_by_ref_test.py

```diff
@@ -1,145 +1,358 @@
+from typing import List, Tuple
+
+import pytest
+
 from pyteal import *
 
 from .compile_asserts import assert_new_v_old, compile_and_save
 
-# Set the following True, if don't want to run pass-by-ref dependent tests
-OLD_CODE_ONLY = False
+# TODO: remove these skips when the following is fixed: https://github.com/algorand/pyteal/issues/199
```

Comment:
Changes in the file mostly consist of the following:
* introduce the `def test_pass_by_ref_guardrails()` test to ensure that the compiler errors on recursive functions that contain `ScratchVar` arguments
* move most of the previously defined recursive but pass-by-ref cases into `test_pass_by_ref_guardrails()`
* **SKIP** other tests in this file with the caveat that once #199 is addressed, they should be turned again. (currently they are flaky)